foxai/explainer/fullgradeints.py

```diff
@@ -0,0 +1,462 @@
+        return input_tensor
+
+    def compute_saliency(
+        self,
+        input_data: torch.Tensor,
+        target: Optional[Union[torch.Tensor, int]] = None,
+    ) -> torch.Tensor:
+        """Compute FullGrad saliency map.
+
+        The function combines input gradient map with bias layer gradient maps.
+
+        Args:
+            input_data: the input image to the model
+            target: the target class to be explained. If none the argmax class is chosen.
+
+        Returns:
+            silency map
+        """
+        if isinstance(target, int):
+            target = torch.tensor(data=[target] * input_data.shape[0]).to(self._device)
+
+        self.model.eval()
+        input_grad, bias_grad = self.full_gradient_decompose(
+            input_data, target_class=target
+        )
+
+        # Input-gradient * image
+        grd = input_grad * input_data
+        gradient = self._post_process(grd).sum(1, keepdim=True)
+        cam = gradient
+
+        im_size = input_data.size()
+
+        # Aggregate Bias-gradients
+        for i in range(len(bias_grad)):
+
+            # Select only Conv layers
+            if len(bias_grad[i].size()) == len(im_size):
+                temp = self._post_process(bias_grad[i])
+                gradient = F.interpolate(
+                    temp,
+                    size=(im_size[2], im_size[3]),
+                    mode="bilinear",
+                    align_corners=True,
+                )
+                cam += gradient.sum(1, keepdim=True)
+
+        return cam
+
+
+class BaseFullGradientCVExplainer(CVExplainer):
+    """Base FullGradCAM algorithm explainer."""
+
+    @abstractmethod
+    def create_explainer(
+        self, model: torch.nn.Module, image_size: Tuple[int, int, int]
+    ) -> FullGrad:
+        """Create explainer object.
+
+        Args:
+            model: The forward function of the model or any
+                modification of it.
+        Returns:
+            Explainer object.
+        """
+
+    def calculate_features(
+        self,
+        model: torch.nn.Module,
+        input_data: torch.Tensor,
+        pred_label_idx: Optional[int] = None,
+        **kwargs,
+    ) -> torch.Tensor:
+        """Generate features image with Full Gradients algorithm explainer.
+
+        Args:
+            model: The forward function of the model or any
+                modification of it.
+            input_data: Input for which attributions
+                are computed. If forward_func takes a single
+                tensor as input, a single input tensor should be provided.
+            pred_label_idx: Output indices for
+                which gradients are computed (for classification cases,
+                this is usually the target class).
+                If None provided, the argmax of output is chosen for explanation.
+
+                Default: None
+
+        Returns:
+            Element-wise product of (upsampled) GradCAM
```

Comment:
I think something is wrong here - mentioned GradCAM and Guided Backprop and not FullGrad. Can you take a look at this?